cf_xarray/accessor.py

```diff
@@ -1733,21 +1733,23 @@ def guess_coord_axis(self, verbose: bool = False) -> DataArray | Dataset:
         obj = self._obj.copy(deep=True)
         for var in obj.coords.variables:
             if obj[var].ndim == 1 and _is_datetime_like(obj[var]):
-                if verbose:
+                new_attrs = dict(ChainMap(obj[var].attrs, ATTRS["time"]))
+                if verbose and new_attrs != obj[var].attrs:
                     print(
                         f"I think {var!r} is of type 'time'. It has a datetime-like type."
                     )
-                obj[var].attrs = dict(ChainMap(obj[var].attrs, ATTRS["time"]))
+                obj[var].attrs = new_attrs
                 continue  # prevent second detection
 
             for name, pattern in regex.items():
                 # match variable names
                 if pattern.match(var.lower()):
```

Comment:
I think we can be a little more explicit. If it matches the pattern for `X` but the `axis` attribute exists, just skip.

Reply:
I changed it a little. The new checks avoid adding attributes to pre-defined axes and coordinates

Reply:
We could have a variable has the standard name "latitude", but not the "units" string. Currently this code would add the units. I think we should preserve that.

Reply:
isn't that `cf.add_canonical_attributes` though?

Reply:
Good point!

I've been relying on `guess_coord_axis` to do it for me.


Let's add a notice to whats-new saying this "feature" has been removed.

Reply:
done